src/contracts/core/RewardsCoordinator.sol

```diff
@@ -229,35 +238,65 @@ contract RewardsCoordinator is
         onlyWhenNotPaused(PAUSED_REWARD_OPERATOR_SET)
         nonReentrant
     {
-        for (uint256 i = 0; i < rewardsSubmissions.length;) {
+        // Cache starting nonce value for later use.
+        uint256 startingNonce = submissionNonce[msg.sender];
+        for (uint256 i = 0; i < rewardsSubmissions.length; ++i) {
             OperatorSetRewardsSubmission calldata rewardsSubmission = rewardsSubmissions[i];
-            uint256 nonce = submissionNonce[msg.sender];
-            bytes32 rewardsSubmissionHash = keccak256(abi.encode(msg.sender, nonce, rewardsSubmission));
-
+            // Calculate current nonce by adding `i` to starting nonce.
+            uint256 currentNonce = startingNonce + i;
+            // Calculate rewards submission hash.
+            bytes32 rewardsSubmissionHash = keccak256(abi.encode(msg.sender, currentNonce, rewardsSubmission));
+            // Assert operator set is valid/registered for the given avs (msg.sender).
             require(
                 avsDirectory.isOperatorSet(msg.sender, rewardsSubmission.operatorSetId),
                 "RewardsCoordinator.rewardOperatorSetForRange: invalid operatorSet"
             );
+            // Validate rewards submission.
             _validateRewardsSubmission(
                 rewardsSubmission.strategiesAndMultipliers,
-                rewardsSubmission.token,
                 rewardsSubmission.amount,
                 rewardsSubmission.startTimestamp,
                 rewardsSubmission.duration,
                 OPERATOR_SET_MAX_RETROACTIVE_LENGTH,
                 OPERATOR_SET_GENESIS_REWARDS_TIMESTAMP
             );
-
+            // Mark `rewardsSubmissionHash` as valid for avs (msg.sender).
             isAVSRewardsSubmissionHash[msg.sender][rewardsSubmissionHash] = true;
-            submissionNonce[msg.sender] = nonce + 1;
-
-            emit OperatorSetRewardCreated(msg.sender, nonce, rewardsSubmissionHash, rewardsSubmission);
+            // Pull reward `token` of `amount` from avs (msg.sender).
             rewardsSubmission.token.safeTransferFrom(msg.sender, address(this), rewardsSubmission.amount);
-
-            unchecked {
-                ++i;
-            }
+            // Emit event to track reward submission.
+            emit OperatorSetRewardCreated(msg.sender, currentNonce, rewardsSubmissionHash, rewardsSubmission);
         }
+        // Increase avs submission nonce by rewards submissions submitted.
+        submissionNonce[msg.sender] = startingNonce + rewardsSubmissions.length;
+    }
+
+    function rewardOperatorsForPerformance(PerformanceRewardsSubmission calldata rewardsSubmission)
```

Comment:
Any reason why this isn't an array? Assume call data will just get too big? That's fine with me.

Reply:
Do you think we should make the other reward submission functions just structs instead of arrays of structs? cc @8sunyuan

Reply:
It wasn't defined as an array in the notion doc, was curious about that though.

Reply:
Assume we are EigenDA and have 200 operators & 200 scores. What would be the max size of the array?

Reply:
My guess is definitely < 5 elements? But also doing an array is consistent with how we are have the rest of the functions, just unsure if its practical..